week7/bmi_calculator_react/01_acceptance_testing_in_react.md

```diff
@@ -41,62 +50,58 @@ Run:
+    cy.visit('/');
+    cy.get('input#weight').type(90);
+    cy.get('input#height').type(190);
+    cy.get('button#calculate').click();
+    cy.get('p#bmi-message').should('contain', 'You are Normal with a BMI of 24.93')
+  })
+})
 ```
-describe('BMI Converter', () => {
-    before(function() {
-        cy.visit('http://localhost:3000');
-    });
-
-    beforeEach(function() {
-        // Reload the application between tests to reset state
-        cy.reload();
-    });
-
-    it('should display "BMI Converter" text on page', () => {
-        cy.contains('BMI Converter');
-    });
-
-    describe('Metric method', () => {
-        beforeEach(() => {
-            // This before block will be executed prior to each test in this describe block
-            cy.get('select[id="method"]').select('metric')
-            cy.get('input[name="weight"]').type('95')
-            cy.get('input[name="height"]').type('186')
-        })
-
-        it('displays assesment', async () => {   
-            cy.contains('You are Overweight')
-        })
-
-        it('displays BMI value', async () => {   
-            cy.contains('BMI of 27.46')
-        })
-    })
-
-    describe('Imperial method', async () => {
-        beforeEach( async () => {
-            // This before block will be executed prior to each test in this describe block
-            cy.get('select[id="method"]').select('imperial')
-            cy.get('input[name="weight"]').type('200')
-            cy.get('input[name="height"]').type('73')
-        })
-
-        it('displays assesment', async () => {   
-            cy.contains('You are Overweight')
-        })
-
-        it('displays BMI value', async () => {   
-            cy.contains('BMI of 26.38')
-        })
-    })
-}); 
-```
-The important part that you need to understand is what is happening in the `beforeEach` and `before` blocks. The `before` makes sure that the test browser goes to the correct URL to run the tests. The `beforeEach` resets the page (and the input fields) after every test.
 
-The testing of the actual method may change depending on the implementation that you have. As you can see it is pretty straight forward, we select the method from a selector, fill in the inputs with the relevant information and then expect what response is going to be shown on the page.
+This is a very simple test of our calculation, we just use our selectors to input and then press the button to see the message. 
 
 If you need to press a button or interact with the page in some different way then what we show you above, you can go and find that in the [Cypress documentation](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell).
 
+
+## Imperial method
+
+I will show you a test case for the imperial method. Your job will be to make the tests pass:
+
+```js
+// cypress/integration/examples/BMI.spec.js
+
+/// <reference types="Cypress" />
+
+describe('BMI Calculator', () => {
+  it('Calculates BMI in metric',() => {
```

Comment:
please add a before block where u go to the website

Reply:
How is this resolved, I don't understand?